packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -0,0 +1,489 @@
+      const vaultId = (await shortSqueeth.nextId());
+      const normFactor = await controller.getExpectedNormalizationFactor()
+      const mintWSqueethAmount = ethers.utils.parseUnits('30')
+      const mintRSqueethAmount = mintWSqueethAmount.mul(normFactor).div(one)
+      const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+      const scaledEthPrice = ethPrice.div(10000)
+      const debtInEth = mintRSqueethAmount.mul(scaledEthPrice).div(one)
+      const collateralToMint = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+      const squeethPrice = await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 420, true)
+      const collateralToLp = mintWSqueethAmount.mul(squeethPrice).div(one)
+      const flashloanWMintDepositNftParams = {
+        vaultId: 0,
+        wPowerPerpAmount: mintWSqueethAmount.toString(),
+        collateralToDeposit: collateralToMint.toString(),
+        collateralToFlashloan: collateralToMint.toString(),
+        collateralToLp: collateralToLp.toString(),
+        collateralToWithdraw: 0,
+        lpAmount0Min: 0,
+        lpAmount1Min: 0,
+        lpLowerTick: -887220,
+        lpUpperTick: 887220
+      }
+
+      await controllerHelper.connect(depositor).flashloanWMintDepositNft(flashloanWMintDepositNftParams, {value: collateralToLp.add(ethers.utils.parseUnits('0.01'))})
+
+      const vaultAfter = await controller.vaults(vaultId)
+      const tokenIndexAfter = await (positionManager as INonfungiblePositionManager).totalSupply();
+      const tokenId = await (positionManager as INonfungiblePositionManager).tokenByIndex(tokenIndexAfter.sub(1));
+      const position = await (positionManager as INonfungiblePositionManager).positions(tokenId)
+
+      expect(BigNumber.from(vaultAfter.NftCollateralId).eq(tokenId)).to.be.true;
+      expect(position.tickLower === -887220).to.be.true
+      expect(position.tickUpper === 887220).to.be.true
+      expect(vaultAfter.shortAmount.sub(mintWSqueethAmount).lte(1)).to.be.true
+      expect(vaultAfter.collateralAmount.eq(BigNumber.from(0))).to.be.true
+    })
+
+    describe("open short with already overcollateralized vault, mint, LP oSQTH only, deposit LP NFT and withdraw ETH collateral", async ()=> {
+      before(async () => {
+        const normFactor = await controller.getExpectedNormalizationFactor()
+        const mintWSqueethAmountToLp : BigNumber = ethers.utils.parseUnits('35')
+        const mintRSqueethAmount = mintWSqueethAmountToLp.mul(normFactor).div(one)
+        const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+        const scaledEthPrice = ethPrice.div(10000)
+        const debtInEth = mintRSqueethAmount.mul(scaledEthPrice).div(one)
+        const collateralAmount = debtInEth.mul(3)
+  
+        await controller.connect(depositor).mintWPowerPerpAmount(0, mintWSqueethAmountToLp, 0, {value: collateralAmount})  
```

Comment:
confused why we do this here, why do we mint 2x (here and in next block)

Reply:
So in this `describe()`, I'm testing minting wPowerPerp without sending any ETH, LP only oSQTH and use as collateral in an already overcollateralized vault, so in the `before()` I opened a vault that is overcollateralized, and in the `it()` block I minted more wPowerPerp, LPed, deposited NFT into that same vault. Does that make sense, or there is a mistake ?